src/compose/ports.ts

```diff
@@ -139,6 +139,12 @@ export class PortMap {
 			}, []);
 	}
 
+	public static normaliseComposePorts(portMaps: PortMap[]): PortMap[] {
```

Comment:
You could have this accept the raw port info and do the `new PortMap` as part of this function

Reply:
I like this idea actually, makes the interface a lot less clunky - will add.